liquidprompt

```diff
@@ -54,7 +54,8 @@ if test -n "${BASH_VERSION-}"; then
     # Must be used for all strings injected in PS1 that may comes from remote sources,
     # like $PWD, VCS branch names...
     __lp_escape() {
-        ret="${1//\\/\\\\}"
+        local arg="${1//\\/\\\\}"
+        ret="${arg//\$/\\\$}"
```

Comment:
Using a check like this (see [the docs](https://www.gnu.org/software/bash/manual/html_node/The-Shopt-Builtin.html)) works perfect in my testing. And doing the check every time is actually faster than storing a check in a variable and checking the variable later (who knew?)
```bash
$ func() { for i in {1..1000000}; do if (( LP_ENABLE_SHORTEN_PATH )); then :; fi; done; }
$ time func
real   	0m4.375s
user   	0m4.276s
sys    	0m0.082s

$ func() { for i in {1..1000000}; do if shopt -q promptvars; then :; fi; done; }
$ time func
real   	0m3.924s
user   	0m3.877s
sys    	0m0.034s
```
```suggestion
        ret="${1//\\/\\\\}"
        if shopt -q promptvars ; then
            ret="${ret//\$/\\\$}"
            ret="${ret//\`/\\\`}"
        fi
```
This change (amended or squashed to one commit) plus a rebase to latest master seems like all the changes needed to get an approval from me.